src/keys.ts

```diff
@@ -0,0 +1,28 @@
+import { purry } from './purry';
+
+function _keys<T>(object: { [key: string]: T } | ArrayLike<T>) {
+  return Object.keys(object) as string[];
+}
+
+/**
+ * Returns a new array containing the keys of the array or object.
+ * @param object
+ * @signature
+ *    R.keys(object)
+ * @example
+ *    R.keys(['x', 'y', 'z']) // => ['1', '2', '3']
+ *    R.keys({ a: 'x', b: 'y', c: 'z' }) // => ['a', 'b', 'c']
+ *    R.pipe(
+ *      { a: 'x', b: 'y', c: 'z' },
+ *      R.values,
+ *      R.first
+ *    ) // => 'a'
+ * @pipeable
+ * @category Object
+ */
+// data-first
+export function keys<T>(object: { [key: string]: T } | ArrayLike<T>): string[];
```

Comment:
I think a better syntax instead of `{ [key: string]: T }` would be:

`Record<PropertyKey, T>`

`Record` is an alternative way of writing mapped objects, which is sufficient for this case, and keys can be any PropertyKey, which I think expands to `string | number | Symbol`.